### ISSUE TITLE
Items disappearing (mod-crossover-bug)

### ISSUE THREAD

**Issue by Nezraddin** [Minecolonies/minecolonies#202]
### Minecolonies version

1.10.2-0.4.1042 (using the testing-modpack)
### Expected behavior
- Sorting the inventory with Inventory-Tweaks
### Actual behaviour
- I had 1 full stack and 8 extra of oak wood in the lumberjack's inventory
- after using the middle-mouse-button-sorting of Inventory tweaks the inventory got sorted, but the 64+8 oak wood changed into two stacks of 8 oak wood
- after a second time doing the same (middle mouse to sort) one of the two got deleted and left me with only 8 oak wood in the end
### Steps to reproduce the problem
- Putting in oak myself didn't cause this bug, so still waiting for my worker to gather enough wood again so I can see if it is easilly reproduced with the oak-wood the lumberjack puts in.

[edit]
Infinite-ressources was "true", if that is important for this.

another edit:
- One of the cut down trees was one of the high trees of the realistic terrain-generation.
  Not sure if that is important, but in 1.10.2 it's the first time I see one of the sapling grow back to one of the high-growing-trees, so not sure if these trees (wood and saplings) are somehow different to vanilla oak trees.

[edit 3]
Tested it now a few times with oak-wood I chopped down myself (not cheated in for testing):
Pretty much happened every time I had more than 8 oak-wood in the lumberjack's inventory, Inventory-sorting always deleted everything over 8.

**Comment by Kostronor** [comment]
Seems like sorting is interfering with the citizen. Don't really know what to do here other than to say "don't sort him" :D

**Comment by Nezraddin** [comment]
Ah ehm, hope I'm not missunderstood here:
I mean sorting in the lumberjack "building", there the bug appears. The lumberjack-citizen is fine to sort.

Sawmill, that's what I meant. Still getting used what is called what ini the mod. ^^"
(amazing mod btw. :D )

**Comment by Dadadah** [comment]
I can confirm that this is an inventory tweaks and minecolonies crossover bug, perhaps it has to do with a minor difference in the minecolonies' storage system?

**Comment by Raycoms** [comment]
Possible that if you click the inventory tweaks buttons the hut loses the
reference to the actualized inventory.
We maybe should check all x ticks if the inventory we have is still correct.

2016-10-17 23:03 GMT-02:00 Dadadah <notifications@github.com>:

**Comment by Dadadah** [comment]
That wouldn't make sense though, because IIRC InventroryTweaks works by moving an item into your inventory and placing it in the location it's sorted at. Which leads me to believe that minecolonies is having a problem with items being inserted into storage too fast.

**Comment by drakray** [comment]
This bug is also caused when you manually stack incomplete stack.
For exemple, put a 48 Cobblestone in Builder, take a 24 in hand, put 24 on 48, makes 64, let 8 in hand, put 8 beside 64, close and reopen, 64 is 48, 8 is 8, lost balance between 48 and 64

**Comment by Kostronor** [comment]
@drakray really? Good catch, thanks!
Than it is definitely our inventory implementation...

**Comment by drakray** [comment]
no problem, keep up the good work :D

**Comment by mgailp** [comment]
I have also had this happen with manual use. I put 3 stacks of slabs into the Miner's hut & when I shift-clicked fences from my inventory into the hut, all but the smallest stack (21) of slabs disappeared. I checked the miner who was there waiting for them and he did not have him.

For me, it seems to always pick the smallest stack, regardless of how many are in that stack.

**Comment by Raycoms** [comment]
Can you guys check if this happens with the newest version on curse? 
(The v19)

**Comment by mgailp** [comment]
Well, part of the prob is that I;m not on the right MC version, LOL. I am on the MineColonies TestPack v 0.4.82 . While I told it to run MC 1.8 when I started, it loaded 1.10. The highest Curse is offering me is v 0.4.84. I updated it to that, as I'm not sure what version would be v19. I'll let you know if that helps. I'll also see if I can get it to run MC 1.8 & if it also does it there, Back in a couple days.
